src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -628,42 +621,47 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, workspaceDAO:
+                  //Attempt to resolve method inputs and submit the workflows to Cromwell, and build the submission status accordingly.
+                  val submittedWorkflows = jobEntities.map(e => submitWorkflow(workspaceContext, methodConfig, e, wdl, userInfo.authCookie, txn))
+                  val newSubmission = Submission(submissionId = UUID.randomUUID().toString,
+                    submissionDate = DateTime.now(),
+                    submitter = userInfo.userId,
+                    methodConfigurationNamespace = methodConfig.namespace,
+                    methodConfigurationName = methodConfig.name,
+                    submissionEntity = AttributeEntityReference(entityType = submission.entityType, entityName = submission.entityName),
+                    workflows = submittedWorkflows collect { case Right(e) => e },
+                    notstarted = submittedWorkflows collect { case Left(e) => e },
+                    status = if (submittedWorkflows.forall(_.isLeft)) SubmissionStatuses.Done else SubmissionStatuses.Submitted)
+
+                  if (newSubmission.status == SubmissionStatuses.Submitted) {
+                    submissionSupervisor ! SubmissionStarted(workspaceName, newSubmission, userInfo.authCookie)
+                  }
+
+                  submissionDAO.save(workspaceContext, newSubmission, txn)
+                  RequestComplete(StatusCodes.Created, newSubmission)
                 }
+              }
+            }
+          }
+        }
+      }
+    }
 
-                //Verify the type of all job entities matches the type of the method configuration.
-                if (!jobEntities.forall(_.entityType == methodConfig.rootEntityType)) {
-                  return RequestComplete(StatusCodes.BadRequest, "Entities " + jobEntities.filter(_.entityType != methodConfig.rootEntityType).map(e => e.entityType + ":" + e.name) +
-                    "are not the same type as the method configuration requires")
-                }
+  def validateSubmission(workspaceName: WorkspaceName, submission: SubmissionRequest): PerRequestMessage =
+    dataSource inTransaction { txn =>
+      withWorkspaceContextAndPermissions(workspaceName, GCSAccessLevel.Write, txn) { workspaceContext =>
+        withMethodConfig(workspaceContext, submission.methodConfigurationNamespace, submission.methodConfigurationName, txn) { methodConfig =>
+          withEntity(workspaceContext, submission.entityType, submission.entityName, txn) { entity =>
+            withMethod(workspaceContext, methodConfig.methodRepoMethod.methodNamespace, methodConfig.methodRepoMethod.methodName, methodConfig.methodRepoMethod.methodVersion, userInfo.authCookie) { agoraEntity =>
+              withWdl(agoraEntity) { wdl =>
+                val (validatedInputs, failedInputs) = validateSubmissionEntities(submission, workspaceContext, entity, methodConfig.rootEntityType)
```

Comment:
I am getting confused by naming... validatedInputs are really the workflow entities, correct?